full-moon/src/tokenizer.rs

```diff
@@ -940,6 +821,69 @@ impl fmt::Display for TokenizerError {
 
 impl std::error::Error for TokenizerError {}
 
+impl From<peg::str::LineCol> for Position {
+    fn from(location: peg::str::LineCol) -> Position {
+        Position {
+            bytes: location.offset,
+            line: location.line,
+            character: location.column,
+        }
+    }
+}
+
+struct TokenCollector<'input> {
+    result: Vec<Token<'input>>,
+}
+
+// Collector
+impl<'input> TokenCollector<'input> {
+    fn new() -> Self {
+        Self { result: Vec::new() }
+    }
+    fn push(
```

Comment:
Have you `cargo fmt`'d this? I'd think this would separate the new lines properly.